### PR TITLE
veax-void-proposal-attempt1

### PR DESCRIPTION
**Pull request by ivanvoid**
---
A good PR should include the following parts:

- [x] A clear title
- [ ] A meaningful message for PR, as well as its commits
- [ ] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [ ] excluding any irrelevant files, such as binaries, text files, or dot files
- [ ] passing tests/CI

### CODE REVIEW

File: proposal/veax-void/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# :pig: PIGS :pig:
+---
+
+Readme for PIGS or Pixelwise Image Graph Segmentation project developmet.
+
+## Basic information
+[Official PIGS repo](https://github.com/veax-void/nsdhw_21sp/tree/veax-void-proposal)
```
Comment (yungyuc) [inline proposal/veax-void/README.md]: The project repository should be different from your clone of the homework one.

File: proposal/veax-void/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# :pig: PIGS :pig:
+---
+
+Readme for PIGS or Pixelwise Image Graph Segmentation project developmet.
```
Comment (yungyuc) [inline proposal/veax-void/README.md:2]: I like the name!

File: proposal/veax-void/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## Basic information
+[Official PIGS repo](https://github.com/veax-void/nsdhw_21sp/tree/veax-void-proposal)
+
+Fast package for image segmentatioin using graphs, like your dad use to make! 
+
+## Problem to solve
+
+Image processing, especially segmentation tasks are really heavy on computations, but also provide a lot of value.
```
Comment (yungyuc) [inline proposal/veax-void/README.md]: Please be specific on what value it provides.

File: proposal/veax-void/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+I propose to use fast cpp code for make computation of relations in the graph more efficient.  
+
+**TODO**: add math and algorithm
+
+## Perspective users
+All people who use image data representations mostly. But this package for programmers who want to integrate fast graph segmentation in to their projects.
+
+## System architecture
+
+**TODO**: Make a system dataflow, add charts!
```
Comment (yungyuc) [inline proposal/veax-void/README.md]: Add system architecture.  It may or may not need a chart.

File: proposal/veax-void/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+## API
+**TODO**: Add API
```
Comment (yungyuc) [inline proposal/veax-void/README.md]: In the beginning, the API description may be very high-level.

File: proposal/veax-void/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+## Infrastructure
+**TODO**: TESTS, DOCS, version control, CI/CD
```
Comment (yungyuc) [inline proposal/veax-void/README.md]: You can itemize what you want to test.

File: proposal/veax-void/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+
+## Schedule
+today: 28/03 Days to go: 70  
+- 5 days: Research
+- 30 days: Development
```
Comment (yungyuc) [inline proposal/veax-void/README.md]: A good starting point.  You may add some sub-bullets on what you plan to develop.  Also, put dates on each of the milestones.

File: proposal/veax-void/README.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+- 10 days: Testing
```
Comment (yungyuc) [inline proposal/veax-void/README.md]: Testing should not wait after development.

**Review by yungyuc — CHANGES_REQUESTED** [review]

**Comment by yungyuc** [comment]
@veax-void please update the proposal itself as soon as possible.

**Comment by ivanvoid** [comment]
> @veax-void please update the proposal itself as soon as possible.

Updated! Added fixed readme and  added first prototype.

File: proposal/veax-void/README.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+Now you can access:
+- **graph.weights** - graph weights
+- **graph.degree** - current graph degree
+- **graph.ground_node** - current ground node
+- **graph.x** - current solved values 
+- **graph.classes** - class of each pixel for current solution
+
+## Infrastructure
+Tests list:
+- [ ] Memory leakage
```
Comment (yungyuc) [inline proposal/veax-void/README.md:43]: Memory leak is hard to detect.  To detect memory error, you need to have a custom memory manager, which you may start with.

File: proposal/veax-void/README.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+## Schedule
+Start date: **28.03.2021**  
+Deadline: **07.06.2021**  
+Last update: **04.10.2021**  
+Days to go: **58**  
+
+Plan:
+> - **deadline**: taks
+>- **04.04.21**: Research
+>- **23.05.21**: Development / Testing
```
Comment (yungyuc) [inline proposal/veax-void/README.md:60]: Adding notes for what you want to test will help your development.

**Review by yungyuc — COMMENTED** [review]
LGTM.  I will merge the PR once finish grading all proposals.

**Review by yungyuc — APPROVED** [review]

**Pull request merged**